<h1>ACTIVITY 2</h1>
<h3>Oira, Jhon Wayne L.</h3>
<h3>BSIT-3A</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
from scipy.stats import f_oneway 
warnings.filterwarnings('ignore')

<h1>IMPORTING THE DATASETS</h1>

In [2]:
accidents_data = pd.read_csv("data\\uk_road_accident.csv")

In [3]:
accidents_data

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

<h1>DESCRIBING THE DATASETS</h1>

In [4]:
accidents_data.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [5]:
accidents_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [6]:
accidents_data.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h1>How many accidents are there in total?</h1>

In [7]:
accidents_data.shape[0]

660679

<h3>INSIGHT:<br><br>1.Based on the data, there are total of (660679)Six Hundred Sixty Thousand Six Hundred Seventy Nine Accidents.</h3>

<h1>How many accidents are there as describe in Accident Severity?</h1>

In [8]:
accidents_data['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<h3>INSIGHTS:<br><br>2. Out of all accidents recorded, 563,801 cases are classified as Slight Severity, indicating that most incidents are less severe.<br>3. There are 88,217 Serious Severity cases, which is considerably lower than Slight Severity but still highlights a significant number of severe incidents that may need preventive measures.<br>4.Only 8,661 cases are Fatal Severity, showing that while fatal accidents are rare compared to others, they still represent critical events that need urgent attention for safety improvements. </h3>

<h1>What is the most severe case in the data provided between Slight, Serious, and Fatal Severity?</h1>

In [9]:
accidents_data["Accident_Severity"].mode()[0]

'Slight'

<h3>INSIGHT:<br><br>5.The data indicates that Slight Severity is the common type of Accidents Severity that happened mostly.</h3>

<h1>In which area do accidents always happen?</h1>

In [10]:
accidents_data["Urban_or_Rural_Area"].mode(0)

0    Urban
Name: Urban_or_Rural_Area, dtype: object

<h3>INSIGHT:<br><br>6. Urbans have more accidents overall, but countryside accidents are often deadlier.</h3>

<h1>What Light Condition do accidents usually occurs?</h1>

In [11]:
accidents_data["Light_Conditions"].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<h3>INSIGHTS:<br><br>7. The largest number of accidents (484,880 cases) happen during daylight, likely due to heavy traffic and increased road activity during the day.<br>8. There are 129,335 accidents that occurred at night even with lights lit, showing that lighting alone does not fully prevent accidents.<br>9. 37,437 accidents happened in areas with no lighting, which emphasizes the danger of poor visibility and the need for infrastructure improvements.<br>10. 6,484 accidents happened where lighting conditions were unknown, highlighting possible data collection issues that should be addressed for better analysis.<br>11. With 2,543 cases, accidents in unlit areas are rare but indicate situations where maintenance or timely lighting repairs could prevent incidents.</h3>

<h1>How many accidents are there as describe in Road Surface Conditions? </h1>

In [12]:
accidents_data["Road_Surface_Conditions"].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<h3>INSIGHTS:<br><br>12.A total of 447,821 accidents happened on dry road surfaces, suggesting that high traffic volume rather than poor conditions is the main factor in these cases.<br>13. 186,708 cases happened on wet or damp roads, indicating that rain or moisture increases the risk of road accidents.<br>14. 18,571 accidents occurred on frosty or icy surfaces, showing how slippery conditions pose a major safety risk.<br>15. There are 5,890 accidents on snowy roads, highlighting the importance of proper winter road maintenance.<br>16. With 1,017 accidents, flood over 3cm deep has the lowest count but could have severe outcomes due to vehicle control loss or road damage.</h3>

<h1>Which road surface condition is most frequently reported during accidents?</h1>

In [13]:
accidents_data["Road_Surface_Conditions"].mode()[0]

'Dry'

<h3>INSIGHTS:<br>17. For Road Surface Conditions, Dry Condition is the most reported accidents.</h3>

<h1>What vehicle type has the most record of accident?</h1>

In [14]:
accidents_data["Vehicle_Type"].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

<h3>INSIGHTS:<br><br>18. With 497,992 cases, cars account for the majority of accidents, likely due to their high number on the road compared to other vehicles.<br>19. 34,160 accidents involve small goods vehicles, showing that delivery and transport vehicles are also frequently involved in road incidents.<br>20. 25,878 bus/coach accidents highlight the importance of passenger safety in public transportation.<br>21. 25,657 accidents involve larger motorcycles, which may reflect higher speeds and increased vulnerability.<br>22. 15,269 accidents show that smaller motorcycles are also prone to crashes, possibly due to their use by younger or less experienced riders.<br>23. 17,307 accidents for goods vehicles over 7.5 tonnes and 6,096 accidents for goods vehicles between 3.5t–7.5t indicate the need for stricter safety measures for large trucks.<br>24. With 1,976 and 1,947 cases respectively, they are among the lower accident contributors but still require attention due to passenger or rural risk factors.<br>25. Only 197 pedal cycle accidents and 4 horse-related accidents were recorded, making them the least frequent contributors to road incidents.</h3>